webapp/nuxt.config.js

```diff
@@ -107,17 +122,98 @@ module.exports = {
     middleware: ['authenticated'],
     linkActiveClass: 'router-link-active',
     linkExactActiveClass: 'router-link-exact-active',
-    scrollBehavior: () => {
-      return { x: 0, y: 0 }
+    scrollBehavior: (to, _from, savedPosition) => {
+      let position = false
+      // if no children detected and scrollToTop is not explicitly disabled
+      if (
+        to.matched.length < 2 &&
+        to.matched.every(r => r.components.default.options.scrollToTop !== false)
+      ) {
+        // scroll to the top of the page
+        position = {
+          x: 0,
+          y: 0,
+        }
+      } else if (to.matched.some(r => r.components.default.options.scrollToTop)) {
+        // if one of the children has scrollToTop option set to true
+        position = {
+          x: 0,
+          y: 0,
+        }
+      }
+
+      // savedPosition is only available for popstate navigations (back button)
+      if (savedPosition) {
+        position = savedPosition
+      }
+
+      return new Promise(resolve => {
+        // wait for the out transition to complete (if necessary)
+        window.$nuxt.$once('triggerScroll', () => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-15T23:02:40Z" title="Friday, August 16th 2019, 1:02:40 am +02:00">Aug 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -106,17 +121,98 @@ module.exports = {
     middleware: ['authenticated'],
     linkActiveClass: 'router-link-active',
     linkExactActiveClass: 'router-link-exact-active',
-    scrollBehavior: () => {
-      return { x: 0, y: 0 }
+    scrollBehavior: (to, _from, savedPosition) => {
+      let position = false
+      // if no children detected and scrollToTop is not explicitly disabled
+      if (
+        to.matched.length < 2 &&
+        to.matched.every(r => r.components.default.options.scrollToTop !== false)
+      ) {
+        // scroll to the top of the page
+        position = {
+          x: 0,
+          y: 0,
+        }
+      } else if (to.matched.some(r => r.components.default.options.scrollToTop)) {
+        // if one of the children has scrollToTop option set to true
+        position = {
+          x: 0,
+          y: 0,
+        }
+      }
+
+      // savedPosition is only available for popstate navigations (back button)
+      if (savedPosition) {
+        position = savedPosition
+      }
+
+      return new Promise(resolve => {
+        // wait for the out transition to complete (if necessary)
+        window.$nuxt.$once('triggerScroll', () => {
```


I hope this is SSR safe! (although considering it's the `nuxt.config.js` it should be the case)

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-16T07:17:48Z" title="Friday, August 16th 2019, 9:17:48 am +02:00">Aug 16, 2019</time>_

---


One would hope so … 😉